src/TraceEvent/TraceLog.cs

```diff
@@ -8385,9 +8385,20 @@ IEnumerator<TraceModuleFile> IEnumerable<TraceModuleFile>.GetEnumerator()
         internal void SetModuleFileName(TraceModuleFile moduleFile, string fileName)
         {
             Debug.Assert(moduleFile.fileName == null);
+
+            if (moduleFile.fileName == fileName)
+                return;
+            
+            var nameBefore = moduleFile.FilePath; // if fileName == null this property returns "ManagedModule"
             moduleFile.fileName = fileName;
             if (moduleFilesByName != null)
+            {
                 moduleFilesByName[fileName] = moduleFile;
+
+                // we have changed the name, so we also need to remove the old name from the dictionary
+                if (moduleFilesByName.ContainsKey(nameBefore))
```

Comment:
:bulb: There is no need for `ContainsKey` here. `Remove` is a NOP if the key isn't present.

Reply:
@sharwell thanks! I did not know that